### DISCUSSION TITLE
"Jingle Mode" no longer in Song Playback Timeline.

### DISCUSSION THREAD

**Discussion by paulcjkp** [AzuraCast/AzuraCast#5572] [General]
Since updating to the latest rolling release, I have noticed that files within a "Jingle Mode" playlist are no longer appearing under Reports> Song Playback Timeline.  I had asked about this in the discord, and was told that this was reported as a bug, and fixed.  However, I do not see this feature as a bug.

I see the 2 bug reports regarding Jingles (https://github.com/AzuraCast/AzuraCast/issues/5542) and (https://github.com/AzuraCast/AzuraCast/issues/5533).  I am in complete agreement that these are bugs, and needed to be fixed.  Jingles should not appear on the player/player history, webhooks, or in the stats page.  But there needs to be a way to see this on the backend.  Every radio automation program or app I've used has shown the full and complete history in the backend, regardless if it's songs, jingles, ads, news, etc., while only pushing music to the frontend.

I am having some issues with song based playlist, that is supposed to play one track at specific times.  Sometimes it runs, sometimes it doesn't.  I can easily verify if it ran by going to Reports> Song Playback Timeline, and looking for the time it should be played.  My "Jingle Mode" playlists are setup the same way, play one track at specific times.  Unfortunately, I can no longer verify if it ran via the Song Playback Timeline.

I know I can access the Liquidsoap Logs, and search through all that code for the jingles, but it appears Azuracast only shows the last 24hrs, or since last restart.  Even if I had access to the full logs, I still need to search through a ton of code trying to find them.  Furthermore, lets say I want to run ads, using "Jingle Mode" playlists.  The client wants to know a) what time did the ad run, and b) how many people listened.  I can find when, but not how many people listened.

Is there any way we can get jingles to appear on the Song Playback Timeline again?  Even as an option to turn on/off for those who don't want to see this?

**Comment by BusterNeece** [comment]
@paulcjkp This is a perfectly valid point; it's almost impossible to reliably debug jingle playlist playback if they aren't actually listed in the song playback timeline.

I've updated the processing/feedback code so that it will log jingle songs in the playback history, but won't display them to the public in either the now-playing or history widgets. For added convenience, jingles will display in the timeline with a slightly muted text color to visually distinguish them from tracks that are shown to listeners.

Hopefully this will be a best-of-both-worlds scenario!

**Reply by paulcjkp** [reply]
Thank you!
